File: dev-page/src/main/java/org/example/page/task/signin/AssemblerSignIn.java
package org.example.page.task.signin;

import org.example.core.page.cookie.ICookie;
import org.example.core.page.presentation.AbstractIPage;

public class AssemblerSignIn {
    public static AbstractIPage getPage(ICookie cookie) {
        final var request = new Request();
        final var service = new Service(request, cookie);
        final var controller = new Controller(service);
        return new Page(controller);
    }
}

File: dev-page/src/main/kotlin/org/example/page/task/signin/AssemblerSignIn.kt
package org.example.page.task.signin

import org.example.core.page.cookie.ICookie
import org.example.core.page.presentation.AbstractIPage

object AssemblerSignIn {
    fun getPage(cookie: ICookie?): AbstractIPage {
        val request = Request()
        val service = Service(request, cookie)
        val controller = Controller(service)
        return Page(controller)
    }
}
